Data Structures/BinaryTree/BinaryTree.java

```diff
@@ -0,0 +1,316 @@
+package Algos;
```

Comment:
1. Create directory structure ```Data Structures/Trees/<Type of Tree>/<language>/<filename>```
2. Create language folder ```java``` and place the file in it. Prefer ```snake_case``` for file names.